pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/NullValueTransformer.java

```diff
@@ -41,6 +45,14 @@ public NullValueTransformer(Schema schema) {
         }
       }
     }
+
+    if (tableConfig.getValidationConfig().isNullTimeColumnHandlingEnabled() && schema.getTimeFieldSpec() != null && schema.getTimeFieldSpec().getIncomingGranularitySpec() != null) {
```

Comment:
Use `schema.getSpecForTimeColumn(timeColumn)` to read the field spec

Reply:
Same thing here. I think we can use `getTimeFieldSpec`. `getSpecForTimeColumn` is used to get `DATE_TIME` column.